Hivenet Electronic Warfare + Black K'ois Tweaks

RustingWithYou opened the pull request:
At the request of the Vaurca team, adds a fuckton of Hivenet verbs, as well as making several changes designed for more interesting Vaurca antagonist gimmicks and future events. There is a lot of content, so I'll go through it all here.

**Black K'ois**
- Black k'ois mycosis now has a fifth stage, where the victim becomes basically a discount zombie - they lose the ability to use complex items and feel pain, and they can infect others with a new verb by grabbing them. Stage 5 mycosis also lets the victim control when they cough up spores. 

- Black k'ois now does increasing amounts of brain scarring based on stage when removed. Once at stage 5, removing black k'ois mycosis will kill the host. 

- Black k'ois no longer contains phoron. You can feed it to people to infect them without killing them.

**Lii'dra/Antagonist Vaurca**

- Adds a new language, "Lii'draic Hivemind". This is not available outside of adminbus, but it basically acts as an encrypted Hivenet for Lii'dra + k'ois zombies. Normal Vaurca have a 1% chance to intercept Lii'draic Hivemind messages, increasing based on subspecies and brood. 

- Adds an admin-only non-canon 'Lii'drafier', which will give a Vaurca the Lii'draic Hivemind and infect a non-Vaurca with black k'ois. This is designed for ease of setup for events and approved antag gimmicks. 

- Adds a 'Hivenet Electronic Warfare Suite' implant to the uplink, which gives several verbs designed for messing with other Vaurca such as shocking them, temporarily preventing them from speaking on the Hivenet, and accessing them as though they were a camera. More on that is provided below.

**Hivenet Administrative Verbs**
Adds a new "administrative neural socket", given to Vaurca Breeders. This socket grants several verbs for controlling the Hivenet. It also, purely for fluff, has a light indicating whether the remote cast system has been triggered. All of these verbs are only available to Breeders.

- Hivenet Mute - prevents a Vaurca from speaking on the Hivenet. Always works on unprotected Vaurca of your own Hive, and has a chance to work on those of other Hives dependent on their augments and your brood, with Mouv Ta getting a bonus. Can be undone by using the verb on them again.
- Hivenet Ban - removes the Hivenet language from a Vaurca, cutting them off completely. Always works on unprotected Vaurca of your own Hive, and has a chance to work on those of other Hives dependent on their augments and your brood, with Mouv Ta getting a bonus. Can be undone by using the verb again.
- Hivenet Void - a last resort and only done in severe circumstances, destroys a Vaurca's neural socket, completely cutting them off from the Hivenet and VR permanently. Cannot be undone.
- Access Hivenet User Senses - basically, uses a Vaurca as a camera. Unprotected Vaurca of your own Hive do not get a choice, while other Vaurca must consent. Mouv Ta can attempt to forcefully hijack another Vaurca, but with a low chance of success.
- Neural Shock - Inflicts a small amount of pain and a smaller amount of brain damage on a Vaurca of your Hive. 
- Hivenet Emergency Transmission - Sends a message to CCIA. Flavored as an emergency Hivenet broadcast sent to one of the three Hives.

**Hivenet Electronic Warfare**
Several verbs and augments have been added to regular Vaurca, Breeders and antagonist Vaurca to allow for some Hivenet-based conflict. Where each of these verbs is available is listed.

- Encrypt Hivenet Message - available to Ta and with the Electronic Warfare implant. Lets you set a key to encrypt your Hivenet messages with. If another Vaurca has set their decryption key to the same, they can understand you, otherwise it shows up as a simple "surge of encrypted data"
- Receive Encrypted Hivenet - available to everyone. This lets you set a key for receiving encrypted messages sent using the above verb.
- Decrypt Hivenet - available to Ta and with the Electronic Warfare implant. Has a 5% chance to give you a random encryption key which is currently in-use. 
- Hijack User Senses - available with the Electronic Warfare implant. Weaker version of the admin verb. User gets an option to ask permission or not - if the hijack is forced, it has a 30% chance to succeed, which will let you use them as a camera without their knowledge. On a failure, it will cause the user pain and let the target know the area that the attack came from.
- Neural Shock - available with the Electronic Warfare implant. Similar to the admin verb, but with a 30% chance of success and without identifying the attacker. On a failure, it causes the user pain and lets the target know the area that the attack came from.
- Disrupt Hivenet User - available with the Electronic Warfare implant. It has a 30% chance of success, and acts as a weaker Hivenet Mute, preventing the target from speaking for 10 minutes. On a failure, it reveals the user's location to the target.

Also adds the Hivenet Electronic Warfare and Hivenet Electronic Defense Implants. The latter provides additional defense against Electronic Warfare and Hivenet Admin verbs, while the former also adds several of the Electronic Warfare verbs. The defense implant is given to Vaurca Breeders, while the warfare implant is available from the uplink. 

An 'advanced' defense implant is also added, which provides total immunity. It is not available to antagonists, and currently is only available to warforms and via adminbus.

RustingWithYou commented:
!review

Generalcamo commented on code/game/objects/items/devices/traitordevices.dm:
```diff
@@ -54,3 +54,34 @@ effective or pretty fucking useless.
+		to_chat(target, SPAN_NOTICE("You have been connected to the Lii'dra Hivemind!"))
+		target.add_language(LANGUAGE_LIIDRA)
+		qdel(src)
+	if(!target.isSynthetic())
```
```suggestion
	else if(!target.isSynthetic())
```

Generalcamo commented on html/changelogs/RustingWithYou - willofthequeens.yml:
```diff
@@ -0,0 +1,46 @@
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Adds several new Hivenet verbs for administrating, covert communication and espionage.
```
```suggestion
  - rscadd: "Adds several new Hivenet verbs for administrating, covert communication and espionage."
```

dessysalta commented:
This is really interesting, I like the amount of depth it adds. I do have a somewhat related question, though--since black k'ois is sort of renown at this point for having an incredibly slow incubation period (about an hour alone just to become a zombie), do you think it would be possible to speed up the progression of symptoms? Would you want to do that?

NonQueueingMatt commented on code/modules/mob/language/outsider.dm:
```diff
@@ -117,3 +117,61 @@
+	if(isvaurca(M))
+		var/interceptchance = 1 //tiny chance for normal bugs to hear a message
+		if(M.species.name == SPECIES_VAURCA_BREEDER) //ta are better at intercepting transmissions
+			if(M.origin.name == "Tupii-K'lax Brood" || M.origin.name == "Vedhra Brood") //experienced in fighting lii'dra
```
Origins should be typechecked, not name checked

NonQueueingMatt commented on code/modules/mob/language/outsider.dm:
```diff
@@ -117,3 +117,61 @@
+		if(M.species.name == SPECIES_VAURCA_BREEDER) //ta are better at intercepting transmissions
+			if(M.origin.name == "Tupii-K'lax Brood" || M.origin.name == "Vedhra Brood") //experienced in fighting lii'dra
+				interceptchance = 15
+			else if(M.origin.name == "Mouv Brood") //electronic warfare specialists
```
Origins should be typechecked, not name checked

NonQueueingMatt commented on code/modules/mob/language/outsider.dm:
```diff
@@ -117,3 +117,61 @@
+			else
+				interceptchance = 10
+		else
+			if(M.origin.name == "Tupii-K'lax Brood" || M.origin.name == "Vedhra Brood")
```
Origins should be typechecked, not name checked

NonQueueingMatt commented on code/modules/mob/language/outsider.dm:
```diff
@@ -117,3 +117,61 @@
+		else
+			if(M.origin.name == "Tupii-K'lax Brood" || M.origin.name == "Vedhra Brood")
+				interceptchance = 5
+			else if(M.origin.name == "Mouv Brood")
```
Origins should be typechecked, not name checked

NonQueueingMatt commented on code/modules/mob/language/station.dm:
```diff
@@ -247,8 +248,20 @@
+		if(istype(player,/mob/abstract/observer) || player == speaker)
 			to_chat(player, msg)
+		else if((src in player.languages && !within_jamming_range(player)) || check_special_condition(player))
+			var/mob/living/carbon/human/H = player
+			var/mob/living/carbon/human/M = speaker
```
this needs typechecking, you don't know for sure if H and M are humans

NonQueueingMatt commented on code/modules/mob/language/station.dm:
```diff
@@ -296,6 +309,17 @@
+		return FALSE
+	else
+		return TRUE
+
+
+
```
random whitespace

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
 	G.tail_storage.open(usr)
+
+//Hivenet Admin
+/mob/living/carbon/human/proc/hivenet_transmit()
+	set name = "Emergency Hivenet Transmission"
+	set desc = "Send a direct Hivenet transmission to your superiors in the Hive. Only to be used in dire circumstances, on pain of severe consequences.."
+	set category = "Hivenet"
+	var/hives = list("Zo'ra", "K'lax", "C'thur")
+
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages))
```
All of the verbs in this file need to check for incapacitation. Most of them also don't have proper spacing between the verb setup (set name, set desc, set category) and the actual code - please add that.

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			var/list/player_fullname = splittext(player.name, " ")
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+		target.remove_language(LANGUAGE_VAURCA)
+		S.banned = TRUE
+		host.last_action = world.time + 1 MINUTES
+	else if(src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			var/list/player_fullname = splittext(player.name, " ")
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+		target.adjustBrainLoss(20)
+		S.Destroy()
+		host.last_action = world.time + 1 MINUTES
+	else if(src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+		to_chat(target, SPAN_WARNING("You feel [src]'s will enter your mind, disabling your socket's ability to speak. Though the voices of your fellow Vaurcae echo still, you cannot speak to them."))
+		S.muted = TRUE
+		host.last_action = world.time + 5 MINUTES
+	else if(src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting Hivenet decryption!"))
+	if(src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+			reset_view(target)
+			return
+		if(choice == "Deny")
+			if(src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			var/list/player_fullname = splittext(player.name, " ")
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+		target.adjustBrainLoss(10)
+		host.last_action = world.time + 5 MINUTES
+		return
+	if(src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+	var/list/selected_vaurca = list()
+	var/max = 3
+	var/obj/item/organ/internal/vaurca/neuralsocket/admin/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			var/list/player_fullname = splittext(player.name, " ")
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || src.origin.name == "Mouv Brood")
```
Origins must be typechecked

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+			if(player_surname == surname || src.origin.name == "Mouv Brood")
+				LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src) //can't ban/mute/void yourself
+	var/mob/living/carbon/human/target = input(src, "Select a Vaurca to mute.", "Hivenet Mute") as null|anything in available_vaurca
```
There's a potential issue here in that you can use this on antag offship vaurca, even Lii'dr, which I don't think should be the case, at least for the latter

RustingWithYou replied:
> There's a potential issue here in that you can use this on antag offship vaurca, even Lii'dr, which I don't think should be the case, at least for the latter

This is why the shield implant was added to the uplink

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting another Hivenet adminisrative action!"))
+	if(target_surname == surname)
+		if(S.shielded == 1)
```
These are magic numbers. Make it into a define

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,782 @@
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before doing this!"))
+		return
+	for(var/mob/living/carbon/human/player in living_mob_list)
```
As before, this works on offship antag vaurca, even Lii'dra

NonQueueingMatt commented on code/modules/organs/subtypes/parasite.dm:
```diff
@@ -234,11 +247,56 @@
+			owner.max_stamina = 500 //you are a discount zombie, you run good
+			add_verb(owner, /mob/living/carbon/human/proc/kois_cough)
+			add_verb(owner, /mob/living/carbon/human/proc/kois_infect)
+			full_zombie = 1
```
boolean

NonQueueingMatt commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,191 @@
 	robotic = ROBOTIC_MECHANICAL
-
-/obj/item/organ/vaurca/neuralsocket/process()
+	var/shielded = 0 //0 = no shield, 1 = normal shield, 2 = immune
```
Make these into defines

RustingWithYou commented:
> > Hivenet Emergency Transmission - Sends a message to CCIA. Flavored as an emergency Hivenet broadcast sent to one of the three Hives.
> 
> not sure how i feel about a function that amounts to an implanted fax machine. antagonists being able to block access to fax machines should be possible, even if its hard with the amount that exist, and this would circumvent that, albeit only for vaurca breeders.

This is blocked by radio jammers, the same as regular Hivenet messages

connorjg1 commented:
> > Hivenet Emergency Transmission - Sends a message to CCIA. Flavored as an emergency Hivenet broadcast sent to one of the three Hives.
> 
> not sure how i feel about a function that amounts to an implanted fax machine. antagonists being able to block access to fax machines should be possible, even if its hard with the amount that exist, and this would circumvent that, albeit only for vaurca breeders.

It's blocked by radio-jammers and is closer in equivalence to an emergency message that a command member can do. It's already been run past CCIA and approved, so no issue on that front.

NonQueueingMatt commented on code/game/objects/items/devices/traitordevices.dm:
```diff
@@ -54,3 +54,34 @@ effective or pretty fucking useless.
+	if(!istype(target))
+		to_chat(user, SPAN_NOTICE("[target] is not a valid target!"))
+		return
+	if(all_languages[LANGUAGE_LIIDRA] in target.languages || target.internal_organs_by_name["blackkois"])
+		to_chat(user, SPAN_NOTICE("[target] is already part of the Lii'dra Hivemind!"))
```
this needs to return

NonQueueingMatt commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,191 @@
+		if(!remote_cast)
+			owner.visible_message(SPAN_WARNING("[owner]'s corpse spasms suddenly, legs twitching and antennae moving wildly as something hums beneath [owner.get_pronoun("his")] exoskeleton. After a moment, [owner.get_pronoun("he")] goes still."))
+			icon_state = "admin_socket_on"
+			desc = "The single most important organ for a Vaurca, able to copy their mind into their Virtual Reality Afterlife upon death. \
+			This one appears to be the far rarer administrative model including encrypted Hivenet access codes and an Emergency Remote Cast System. These are almost never found on any Vaurca Bioform except the Ta.\
+			The Emergency Remote Cast light is green, indicating it has been triggered."
```
this can just be `desc += "The emergency remote cast blahblahblah."`

NonQueueingMatt commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,191 @@
+	if (is_broken())
+		if (all_languages[LANGUAGE_VAURCA] in owner.languages)
+			owner.remove_language(LANGUAGE_VAURCA)
+			to_chat(owner, "<span class='warning'>Your mind suddenly grows dark as the unity of the Hive is torn from you.</span>")
```
use SPAN_X() macros

NonQueueingMatt commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,191 @@
+	desc = "The single most important organ for a Vaurca, able to copy their mind into their Virtual Reality Afterlife upon death. \
+	This one appears to be the far rarer administrative model including encrypted Hivenet access codes and an Emergency Remote Cast System. These are almost never found on any Vaurca Bioform except the Ta.\
+	The Emergency Remote Cast light is red, indicating it has not been triggered."
+	var/remote_cast = 0 //get out of death free card
```
boolean, make it into TRUE/FALSE

NonQueueingMatt commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,191 @@
+		S.shielded = SOCKET_UNSHIELDED
+		return
+	to_chat(user, SPAN_NOTICE("You activate the [src], shielding your neural socket against outside attack."))
+	switch(fullshield)
+		if(TRUE)
+			S.shielded = SOCKET_FULLSHIELDED
+		if(FALSE)
+			S.shielded = SOCKET_SHIELDED
```
```
if(fullshield)
	S.shielded = SOCKET_FULLSHIELDED
else
	S.shielded = SOCKET_SHIELDED
```

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
```
Lots of span tags here that are inconsistent with the rest, make them into macros

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+/mob/living/carbon/human/proc/hivevoid() //Destroys a Vaurca's neural socket.
+	set name = "Void Hivenet User"
+	set desc = "Permanently sever a Vaurca of your own Hive from the Hivenet, destroying their neural socket. Mouv Ta may attempt to use this on Vaurca of another Hive, though this may have severe diplomatic consequences!"
+	set category = "Hivenet"
+	var/list/available_vaurca = list()
```
missing space

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
+		return
+	if(!istype(src.internal_organs_by_name[BP_NEURAL_SOCKET], /obj/item/organ/internal/vaurca/neuralsocket/admin))
+		to_chat(src, "<span class='warning'>You lack the authority to do this!.</span>")
+		return
```
this is probably better off as a variable on the neural socket

RustingWithYou replied:
Done

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || istype(src.origin, /singleton/origin_item/origin/mouv_b))
+				LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src) //can't ban/mute/void yourself
```
not a lazylist, don't use LAZY macros

`for(var/mob/living/carbon/human/player in (player_list - src))` should work

also, shouldn't this be human_mob_list instead?

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+	if(choice == "Cancel")
+		return
+	msg_admin_attack("[key_name_admin(src)] attempted to void [key_name_admin(target)]'s neural socket! (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[src.x];Y=[src.y];Z=[src.z]'>JMP</a>)",ckey=key_name(src),ckey_target=key_name(target))
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting another Hivenet adminisrative action!"))
```
this check is in the wrong place + you need to add a return: it needs to be at the very start of the proc, otherwise it's pointless to get all the way down here - it's sending out a log that you did a thing but then it's erroring out

you also typo'd administrative

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		target.adjustHalLoss(40)
+		target.flash_pain(40)
+		target.adjustBrainLoss(20)
+		S.Destroy()
```
`qdel()`, not Destroy. You should never call Destroy manually

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+				target.adjustHalLoss(40)
+				target.flash_pain(40)
+				target.adjustBrainLoss(20)
+				S.Destroy()
```
ditto

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+			target.adjustHalLoss(40)
+			target.flash_pain(40)
+			target.adjustBrainLoss(20)
+			S.Destroy()
```
ditto

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+/mob/living/carbon/human/proc/hivemute() //Prevents a Vaurca from speaking Hivenet, though they can still hear it
+	set name = "Mute Hivenet User"
+	set desc = "Prevent a Vaurca of your own Hive from speaking on the Hivenet, though they can still hear it. Use this ability again to unmute them. Mouv Ta may attempt to use this on Vaurca of another Hive, though this may have severe diplomatic consequences!"
+	set category = "Hivenet"
+	var/list/available_vaurca = list()
```
space

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || istype(src.origin, /singleton/origin_item/origin/mouv_b))
+				LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src) //can't ban/mute/void yourself
```
as explained earlier

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+			src.adjustHalLoss(15)
+			src.flash_pain(15)
+			host.last_action = world.time + 5 MINUTES
+			return
```
bunch of pointless returns here

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+
+//Hivenet Electronic Warfare
+/mob/living/carbon/human/proc/hivenet_recieve()
+	set name = "Recieve Encrypted Hivenet"
```
Receive* Encrypted Hivenet Key maybe? it's not so clear from the name

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+
+/mob/living/carbon/human/proc/hivenet_encrypt()
+	set name = "Set Hivenet Encryption"
+	set desc = "Set an encryption key for the Hivenet. If someone has this key set to recieve, they will understand you. Use this verb again to reset the key."
```
receive*

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+/mob/living/carbon/human/proc/hivenet_encrypt()
+	set name = "Set Hivenet Encryption"
+	set desc = "Set an encryption key for the Hivenet. If someone has this key set to recieve, they will understand you. Use this verb again to reset the key."
+	set category = "Hivenet"
+	var/obj/item/organ/internal/vaurca/neuralsocket/S = src.internal_organs_by_name[BP_NEURAL_SOCKET]
```
space

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
```
span macro

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		to_chat(src, SPAN_WARNING("You must wait before attempting Hivenet decryption!"))
+	if(istype(src.origin, /singleton/origin_item/origin/mouv_b))
+		decryptchance = 15
+	for(var/mob/living/carbon/human/player in player_list)
```
as explained earlier, the human_mob_list stuff

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
+		return
+	if(stat!=CONSCIOUS)
```
space out the operation properly

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		remoteview_target = null
+		reset_view(0)
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting another Hivenet action!"))
```
this needs to return, and should be at the start of the proc, right below the stat check

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting another Hivenet action!"))
+	for(var/mob/living/carbon/human/player in living_mob_list)
```
just check human list instead of living mob

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			if(player.stat != CONSCIOUS)
+				continue
+			LAZYADD(available_vaurca, player)
```
available_vaurca is not a lazylist, don't use lazy macros

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+			if(player.stat != CONSCIOUS)
+				continue
+			LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src) //you already have the supernatural ability to see through your own eyes
```
available_vaurca is not a lazylist, don't use lazy macros

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		to_chat(src, SPAN_NOTICE("You extend your mind into the Hivenet, accessing [target]'s senses. Use this verb again to cancel."))
+		remoteview_target = target
+		reset_view(target)
+		return
```
no need to return here

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+						to_chat(target, SPAN_WARNING("[src] attempted to hijack your senses, but your countermeasures repelled them!"))
+						remoteview_target = null
+						reset_view(0)
+						host.last_action = world.time + 5 MINUTES
```
The cooldown needs to be set before you get to this big if, otherwise you can kind of spam someone with popups

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !host)
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
```
inconsistent spans

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || istype(src.origin, /singleton/origin_item/origin/mouv_b))
+				LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src) //can't shock yourself
```
not a lazy list

also this codeblock is repeated a lot, maybe make it into a proc

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		src.flash_pain(15)
+		return
+	if(target_surname == surname)
+		to_chat(src, SPAN_WARNING("You lash out over the Hivenet, delivering a neural shock to [target]!"))
+		to_chat(target, SPAN_DANGER("You feel [src]'s will strike out at you, pain burning inside your head!"))
+		target.adjustHalLoss(15)
+		target.flash_pain(15)
+		target.adjustBrainLoss(10)
+		host.last_action = world.time + 5 MINUTES
```
this all looks like it could be a separate proc, lots of copypaste here

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		to_chat(src, SPAN_NOTICE("There are no unshielded Vaurcae within range!"))
+		return
+	LAZYADD(available_vaurca, "Finished")
+	while(selected_vaurca.len < max)
```
for loop, please

`for(var/i = 0 to max)`

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		if(!istype(H))
+			break
+		var/obj/item/organ/internal/vaurca/neuralsocket/S = H.internal_organs_by_name[BP_NEURAL_SOCKET]
+		LAZYADD(selected_vaurca, H)
```
this list is pointless

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You cannot attempt a Hivenet action so soon!"))
+		return
+	for(var/mob/living/carbon/human/player in living_mob_list)
```
as before, iterate over the mob list

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		to_chat(src, SPAN_DANGER("Your disruption attempt fails, [target]'s countermeasures repelling your assault! You can feel their defenses spring to life, tracing your location!"))
+		to_chat(target, SPAN_DANGER("You feel a presence attempting to block your neural socket, but your countermeasures repel it! Analysing the attack, you sense that it came from [get_area(src)]!"))
+	to_chat(target, SPAN_DANGER("Suddenly, you feel an attack on your systems, blocking your neural socket's ability to transmit to the Hivenet!"))
+	S.disrupted = 1
```
boolean

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+
+	var/list/available_vaurca
+	var/obj/item/organ/internal/vaurca/neuralsocket/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.stat != CONSCIOUS)
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!istype(host) || !(all_languages[LANGUAGE_VAURCA] in src.languages))
+		to_chat(src, SPAN_WARNING("Your mind is dark, unable to contact the Hive!"))
+		return
+	if(host.disrupted || host.muted)
+		to_chat(src, SPAN_WARNING("Your neural socket is unable to broadcast!"))
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before you can do another Hivenet action!"))
+		return
```
A lot of these checks are copypasted all over these verbs, you should probably make them into a dynamic proc and just reuse that

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before you can do another Hivenet action!"))
+		return
+	for(var/mob/living/carbon/human/player in living_mob_list)
```
as before, human mob list

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting another Hivenet action!"))
+		return
+	for(var/mob/living/carbon/human/player in living_mob_list)
```
iterate over human mob list instead

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
+		return
+	if(stat!=CONSCIOUS)
```
space out the operator

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
```
span class

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+	to_chat(src, SPAN_NOTICE("You extend your mind into the Hivenet, accessing [target]'s senses. Use this verb again to cancel."))
+	remoteview_target = target
+	reset_view(target)
+	return
```
pointless return

NonQueueingMatt commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,837 @@
+	set desc = "Shock a member of your own Hive with a functioning Neural Socket. Deals pain and brain damage. Rare to need to use, most Ta would only do so in the event of a malfunctioning Vaurca or Viax. A Mouv Ta may attempt this against other Hive members, though to do so would be an extreme diplomatic breach."
+	set category = "Hivenet"
+
+	var/list/available_vaurca = list()
```
this needs to not be initialised if you want to use lazy lists

SleepyGemmy commented on code/game/objects/items/devices/traitordevices.dm:
```diff
@@ -54,3 +54,35 @@ effective or pretty fucking useless.
 		icon_state = "battererburnt"
+
+/obj/item/device/liidrafier //adminspawn/event injector, designed to easily make someone a lii'dra zombie/actual lii'dra
+	name = "Lii'drafication Injector"
+	icon_state = "animal_tagger1"
+	desc = "Use this single-use injector on a Vaurca to grant them access to the Lii'dra Hivenet. Use it on an organic non-Vaurca to infect them with black k'ois. This is an OOC item, do not let anyone see it!"
```
```suggestion
	name = "Lii'drafication Injector"
	desc = "Use this single-use injector on a Vaurca to grant them access to the Lii'dra Hivenet. Use it on an organic non-Vaurca to infect them with black k'ois. This is an OOC item, do not let anyone see it!"
	icon_state = "animal_tagger1"
```

SleepyGemmy commented on code/modules/admin/topic.dm:
```diff
@@ -870,6 +870,24 @@
+		if(!input)	return
+		to_chat(src.owner, "You sent [input] to [H] via a secure Hivenet channel.")
+		log_admin("[src.owner] replied to [key_name(H)]'s Hivenet message with the message [input].", admin_key=key_name(src.owner), ckey=key_name(H))
+		to_chat(H, "<span class='info'>You feel something shifting in the Hivenet, an encrypted whisper transmitted directly to your neural socket. Message as follows.</span>")
+		to_chat(H, "<span class='notice'><b>\"[input]\"</b></span>")
+		to_chat(H, "<span class='info'>Message ends.</span>")
```
```suggestion
		to_chat(H, SPAN_INFO("You feel something shifting in the Hivenet, an encrypted whisper transmitted directly to your neural socket. Message as follows."))
		to_chat(H, SPAN_NOTICE(SPAN_BOLD("\"[input]\"")))
		to_chat(H, SPAN_INFO("Message ends."))
```

SleepyGemmy commented on code/modules/mob/language/outsider.dm:
```diff
@@ -117,3 +117,61 @@
+			if(!within_jamming_range(player))
+				to_chat(player, msg)
+
+/datum/language/bug/liidra/check_special_condition(mob/other)
+	var/mob/living/carbon/human/M = other
+	if(!istype(M))
+		return 0
+	if(istype(M, /mob/abstract/new_player))
+		return 0
+	if(within_jamming_range(other))
+		return 0
+	if(M.internal_organs_by_name[BP_NEURAL_SOCKET] && (all_languages[LANGUAGE_LIIDRA] in M.languages)) //replace with Special Liidra Socket later
+		return 1
+	if(M.internal_organs_by_name["blackkois"] && (all_languages[LANGUAGE_LIIDRA] in M.languages))
+		return 1
+	if(isvaurca(M))
+		var/interceptchance = 1 //tiny chance for normal bugs to hear a message
+		if(M.species.name == SPECIES_VAURCA_BREEDER) //ta are better at intercepting transmissions
+			if(istype(M.origin, /singleton/origin_item/origin/tupii_b) || istype(M.origin, /singleton/origin_item/origin/vedhra_b)) //experienced in fighting lii'dra
+				interceptchance = 15
+			else if(istype(M.origin, /singleton/origin_item/origin/mouv_b)) //electronic warfare specialists
+				interceptchance = 20
+			else
+				interceptchance = 10
+		else
+			if(istype(M.origin, /singleton/origin_item/origin/tupii) || istype(M.origin, /singleton/origin_item/origin/vedhra))
+				interceptchance = 5
+			else if(istype(M.origin, /singleton/origin_item/origin/mouv))
+				interceptchance = 10
+		if(prob(interceptchance))
+			return 1
+	return 0
```
are the returns in this datum booleans (i.e. TRUE/FALSE)? if so, please use the macros (`return TRUE` or `return FALSE`).

SleepyGemmy commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -435,6 +435,8 @@
 		return FALSE
 	if (analgesic > 100)
 		return FALSE
+	if (pain_immune)
```
```suggestion
	if(pain_immune)
```

SleepyGemmy commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -863,6 +863,11 @@
+	if(lobotomized)
+		if(!silent)
+			to_chat(src, SPAN_WARNING("You are in no state to use that!"))
+		return 0
```
boolean?

SleepyGemmy commented on code/modules/organs/subtypes/parasite.dm:
```diff
@@ -174,11 +176,17 @@
 		if(prob(5))
-			to_chat(owner, "<span class='warning'>You feel something squirming inside of you!</span>")
+			if(stage < 4)
+				to_chat(owner, "<span class='warning'>You feel something squirming inside of you!</span>")
```
```suggestion
				to_chat(owner, SPAN_WARNING("You feel something squirming inside of you!"))
```

SleepyGemmy commented on code/modules/organs/subtypes/parasite.dm:
```diff
@@ -207,17 +215,22 @@
 			if(owner.can_feel_pain())
 				to_chat(owner, "<span class='warning'>You feel an unbearable pain in your mind!</span>")
 				owner.emote("scream")
+			else if (full_zombie)
```
```suggestion
			else if(full_zombie)
```

SleepyGemmy commented on code/modules/organs/subtypes/parasite.dm:
```diff
@@ -207,17 +215,22 @@
-			to_chat(owner, "<span class='danger'>You feel something alien coming up your throat!</span>")
-
+			if(stage <= 4)
+				to_chat(owner, "<span class='danger'>You feel something alien coming up your throat!</span>")
```
```suggestion
				to_chat(owner, SPAN_DANGER("You feel something alien coming up your throat!"))
```

SleepyGemmy commented on code/modules/organs/subtypes/parasite.dm:
```diff
@@ -234,11 +247,56 @@
+			add_verb(owner, /mob/living/carbon/human/proc/kois_infect)
+			full_zombie = TRUE
+
 /obj/item/organ/internal/parasite/blackkois/removed(var/mob/living/carbon/human/target)
+	var/obj/item/organ/internal/brain/B = target.internal_organs_by_name[BP_BRAIN]
+	switch(stage)
+		if(1)
+			target.visible_message(SPAN_WARNING("As the black k'ois is cut from [target.name], [target.get_pronoun("he")] spasm[target.get_pronoun("end")] briefly, the black k'ois pulsing in [target.get_pronoun("his")] brain. You can see the scars it has left - it was not a clean surgery, but the damage is minor. [target.get_pronoun("He")] will likely fully recover, given time."),\
+			SPAN_WARNING("They cut at your mind, removing the growth that infests it. Your head aches - but your mind is still intact."))
+			to_chat(target, SPAN_CULT("A shiver passes over your mind, an odd sense of loneliness. But it lasts only a moment, and then it is gone."))
+			B.max_damage = B.max_damage*0.95 //you got it early, good job
+		if(2)
+			target.visible_message(SPAN_WARNING("As the black k'ois is cut from [target.name], [target.get_pronoun("he")] spasm[target.get_pronoun("end")] wildly, the black k'ois embedded in [target.get_pronoun("his")] brain. You can see the scars it has left - though it was not a clean surgery, the damage is not too great. [target.get_pronoun("He")] may well fully recover."),\
+			SPAN_WARNING("They cut at your mind, removing the growth that infests it. Your head hurts, and you find some things hard to recall - but you remember who you are, and you are free from a truly terrible fate. Perhaps that is some comfort."))
+			to_chat(target, SPAN_CULT("Something lingers on you. You are free of the mycosis before it could kill you - and why does that fact make you feel almost sad?"))
+			B.max_damage = B.max_damage*0.75 //bit more scarring but still ok
+		if(3)
+			target.visible_message(SPAN_WARNING("As the black k'ois is cut from [target.name], [target.get_pronoun("he")] spasm[target.get_pronoun("end")] wildly, the black k'ois deeply embedded in [target.get_pronoun("his")] brain. You can see the scars it has left - it was not a clean surgery, and [target.get_pronoun("he")] may not fully recover - but [target.get_pronoun("he")] will live, at least."),\
+			SPAN_WARNING("They cut at your mind, removing the growth that infests it. Your head hurts, and you find some things hard to recall - but you remember who you are, and you are free from a truly terrible fate. Perhaps that is some comfort."))
+			to_chat(target, SPAN_CULT("A part of you misses the feeling of it, pulsing inside your skull like a friend, a constant companion. You have survived - but you cannot help but feel empty."))
+			B.max_damage = B.max_damage*0.65 //ok, now you have permanent brain damage
+		if(4)
+			target.visible_message(SPAN_WARNING("As the black k'ois is cut from [target.name], [target.get_pronoun("he")] spasm[target.get_pronoun("end")] wildly, the black k'ois deeply embedded in [target.get_pronoun("his")] brain. You can see the scars it has left - it was not a clean surgery, and [target.get_pronoun("he")] will never fully recover - but [target.get_pronoun("he")] will live, at least.") ...
+			SPAN_WARNING("They cut at your mind, removing the growth that infests it. It's hard to remember what you are, without it. Perhaps, if you think hard enough, you can remember your name."))
+			to_chat(target, SPAN_CULT("You have failed Us. We will find another way. Releasing control."))
+			B.max_damage = B.max_damage*0.45 //yeah your brain is severely messed up man
+		if(5)
+			target.visible_message(SPAN_WARNING("As the black k'ois is cut from [target.name], [target.get_pronoun("he")] spasm[target.get_pronoun("end")] wildly, [target.get_pronoun("his")] brain too damaged to function. A horrible choking sound emerges from [target.get_pronoun("his")] mouth, before it is mercifully silenced as [target.get_pronoun("he")] falls still."),\
+			SPAN_DANGER("You feel the cold of the knife cutting and cutting, but it is too late. There is nothing left of you but the Lii'dra's will, now. A final cut - and then you are granted the only escape that still remains to you."))
+			to_chat(target, SPAN_GOOD("Your function is fulfilled. We have no further need of you. Releasing control."))
+			target.death(FALSE) //too late, there is no brain left to save
 	if(all_languages[LANGUAGE_VAURCA] in target.languages && stage >= 3 && !isvaurca(target))
 		target.remove_language(LANGUAGE_VAURCA)
 		to_chat(target, "<span class='warning'>Your mind suddenly grows dark as the unity of the Hive is torn from you.</span>")
+	if(all_languages[LANGUAGE_LIIDRA] in target.languages && stage >= 3)
+		target.remove_language(LANGUAGE_LIIDRA)
 	removed_langs = 0
+	remove_verb(owner, /mob/living/carbon/human/proc/kois_cough)
+	remove_verb(owner, /mob/living/carbon/human/proc/kois_infect)
 	..()
```
these messages should respect unconsciousness and the tumor being removed (how is the tumor speaking to you if it's removed? unless i'm misunderstanding.), but you may skip this if it's too much work.

RustingWithYou replied:
The idea is that it's sort of in the process of being removed. I imagine it as sort of a surreal dream-state given that, if it's being surgically removed, it's likely the person is unconscious. Having it only appear when conscious would mean that these messages pretty much never get seen, and given that they're 90% flavor I think it's better this way.

As for respecting the tumor being removed, unless I'm misunderstanding the removed() proc, this should be called when the tumor is getting removed - if they don't have one or it's already been removed, it's not an issue.

SleepyGemmy commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,187 @@
+	..()
+
+/obj/item/organ/internal/vaurca/neuralsocket/admin
+	name = "administrative neural socket"
+	icon_state = "admin_socket"
+	desc = "The single most important organ for a Vaurca, able to copy their mind into their Virtual Reality Afterlife upon death. \
+	This one appears to be the far rarer administrative model including encrypted Hivenet access codes and an Emergency Remote Cast System. These are almost never found on any Vaurca Bioform except the Ta."
```
```suggestion
	name = "administrative neural socket"
	desc = "The single most important organ for a Vaurca, able to copy their mind into their Virtual Reality Afterlife upon death. \
	This one appears to be the far rarer administrative model including encrypted Hivenet access codes and an Emergency Remote Cast System. These are almost never found on any Vaurca Bioform except the Ta."
	icon_state = "admin_socket"
```

SleepyGemmy commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,187 @@
+			owner.visible_message(SPAN_WARNING("[owner]'s corpse spasms suddenly, legs twitching and antennae moving wildly as something hums beneath [owner.get_pronoun("his")] exoskeleton. After a moment, [owner.get_pronoun("he")] goes still."))
+			icon_state = "admin_socket_on"
+			desc += "The Emergency Remote Cast light is green, indicating it has been triggered."
+			remote_cast = 1
```
boolean?

SleepyGemmy commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,187 @@
+	name = BP_HIVENET_SHIELD
+	organ_tag = BP_HIVENET_SHIELD
+	parent_organ = BP_HEAD
+	icon_state = "augment-pda" //placeholder
```
```suggestion
	icon_state = "augment-pda"
```
if it's still a placeholder the comment isn't needed, unless there's plans in the near future to implement a replacement.

SleepyGemmy commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,187 @@
+	organ_tag = BP_HIVENET_SHIELD
+	parent_organ = BP_HEAD
+	icon_state = "augment-pda" //placeholder
+	desc = "An augment often seen among Vaurcae specialising in espionage or cyberwarfare operations, this suite of tools is designed to protect a Vaurca's Hivenet connection against hacking, remote access or sabotage."
```
```suggestion
	desc = "An augment often seen among Vaurcae specialising in espionage or cyberwarfare operations, this suite of tools is designed to protect a Vaurca's Hivenet connection against hacking, remote access, and sabotage."
```

SleepyGemmy commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,187 @@
+/obj/item/organ/internal/augment/hiveshield/attack_self(var/mob/living/carbon/user)
+	var/obj/item/organ/internal/vaurca/neuralsocket/S = user.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(!istype(S) && !S.is_broken())
+		to_chat(user, SPAN_WARNING("You require a working neural socket to activate the [src]"))
```
```suggestion
		to_chat(user, SPAN_WARNING("You require a working neural socket to activate \the [src]."))
```

SleepyGemmy commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,187 @@
+		to_chat(user, SPAN_WARNING("You require a working neural socket to activate the [src]"))
+		return
+	if(S.shielded)
+		to_chat(user, SPAN_NOTICE("You deactivate the [src], lowering your neural socket's defenses."))
```
```suggestion
		to_chat(user, SPAN_NOTICE("You deactivate \the [src], lowering your neural socket's defenses."))
```

SleepyGemmy commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,187 @@
+		to_chat(user, SPAN_NOTICE("You deactivate the [src], lowering your neural socket's defenses."))
+		S.shielded = SOCKET_UNSHIELDED
+		return
+	to_chat(user, SPAN_NOTICE("You activate the [src], shielding your neural socket against outside attack."))
```
```suggestion
	to_chat(user, SPAN_NOTICE("You activate \the [src], shielding your neural socket against outside attack."))
```

SleepyGemmy commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,187 @@
+
+/obj/item/organ/internal/augment/hiveshield/advanced
+	name = "advanced hivenet electronic defense suite"
+	desc = "An augment often seen among Vaurcae specialising in espionage or cyberwarfare operations, this suite of tools is designed to protect a Vaurca's Hivenet connection against hacking, remote access or sabotage.\
```
```suggestion
	desc = "An augment often seen among Vaurcae specialising in espionage or cyberwarfare operations, this suite of tools is designed to protect a Vaurca's Hivenet connection against hacking, remote access or sabotage. \
```

SleepyGemmy commented on code/modules/organs/subtypes/vaurca.dm:
```diff
@@ -68,28 +68,187 @@
+
+/obj/item/organ/internal/augment/hiveshield/warfare
+	name = "hivenet electronic warfare suite"
+	desc = "An augment often seen among Vaurcae specialising in espionage or cyberwarfare operations, this suite of tools is designed to protect a Vaurca's Hivenet connection against hacking, remote access or sabotage.\
```
```suggestion
	desc = "An augment often seen among Vaurcae specializing in espionage or cyberwarfare operations, this suite of tools is designed to protect a Vaurca's Hivenet connection against hacking, remote access or sabotage. \
```

SleepyGemmy commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,836 @@
+		to_chat(src, SPAN_DANGER("Your mind is dark, unable to communicate with the Hive."))
+		return
+	if(!istype(src.internal_organs_by_name[BP_NEURAL_SOCKET], /obj/item/organ/internal/vaurca/neuralsocket/admin))
+		to_chat(src, SPAN_WARNING("You lack the authority to do this!."))
```
```suggestion
		to_chat(src, SPAN_WARNING("You lack the authority to do this!"))
```

SleepyGemmy commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -1336,3 +1336,836 @@
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, SPAN_DANGER("Your mind is dark, unable to communicate with the Hive."))
+		return
+	if(!istype(src.internal_organs_by_name[BP_NEURAL_SOCKET], /obj/item/organ/internal/vaurca/neuralsocket/admin))
+		to_chat(src, SPAN_WARNING("You lack the authority to do this!."))
+		return
```
"this is probably better off as a variable on the neural socket" from Matt.
unadressed?

RustingWithYou replied:
> "this is probably better off as a variable on the neural socket" from Matt. unadressed?

I did not notice that because github decided to hide it instead of the 30 resolved ones, hivenet usage checks have been turned into a proc

SleepyGemmy reviewed (changes requested):
still a bunch of unaddressed feedback. look at the unresolved comments from me and Matt.

RustingWithYou commented:
> still a bunch of unaddressed feedback. look at the unresolved comments from me and Matt.

I believe that's all the major feedback addressed, I missed a bunch in hidden conversations.
